ouroboros-consensus/src/Ouroboros/Consensus/Protocol/WithEBBs.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeApplications           #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE UndecidableInstances       #-}
+{-# LANGUAGE UndecidableSuperClasses    #-}
+{-# LANGUAGE ViewPatterns               #-}
+
+module Ouroboros.Consensus.Protocol.WithEBBs
+  ( WithEBBs
+  , HeaderSupportsWithEBB(..)
+  , NodeConfig(..)
+  )
+where
+
+import Data.Proxy (Proxy(..))
+import Ouroboros.Consensus.Protocol.Abstract
+
+class ( SupportedHeader p (HOEHdr hoe)
+      -- , HasHeader (HOEEbb hoe)
+      ) => HeaderSupportsWithEBB p hoe where
+
+  type HOEHdr hoe :: *
+  type HOEEbb hoe :: *
```

Comment:
Hmm, these names are quite cryptic. We won't use them too much, so longer names are not too annoying.

What about `HeaderOfEBB` and `HeaderOfBlock`. Or, if we only import this module qualified as `WithEBBs`, then even `EBB` and `(Regular)Block` might work.

It'd also use `h` instead of `hoe`.